packages/opentelemetry-exporter-datadog/src/transform.ts

```diff
@@ -96,56 +93,106 @@ function createSpan(
   ddSpanBaseContext._spanId = ddSpanId;
   ddSpanBaseContext._parentId = ddParentId;
 
-  // set reserved service and resource tags
-  ddSpanBase.addTags({
-    'resource.name': createResource(span),
-    'service.name': service_name,
-  });
-
   // set error code
+  addErrors(ddSpanBase, span);
+
+  // set span kind
+  addSpanKind(ddSpanBase, span);
+
+  // set span type
+  addSpanType(ddSpanBase, span);
+
+  // set datadog specific env and version tags
+  addDatadogTags(ddSpanBase, span, serviceName, env, version);
+
+  // set sampling rate
+  setSamplingRate(ddSpanBase, span);
+
+  // set span duration
+  setDuration(ddSpanBase, span);
+
+  // mark as finished span so that it is exported
+  ddSpanBaseContext._isFinished = true;
+
+  return ddSpanBase;
+}
+
+function addErrors(ddSpanBase: typeof Span, span: ReadableSpan): void {
   if (span.status && span.status.code && span.status.code > 0) {
     // TODO: set error.msg error.type error.stack based on error events
```

Comment:
Nit: would you mind create an issue in repo so this is not forgotten ?

Reply:
yup can do, i'll add an open issue linking to the spec pr https://github.com/open-telemetry/opentelemetry-specification/pull/697 (which got approved it looks like 🎉 )